
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Make plot highlighting record datasets over time &#8212; Single Cell Studies</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Example analysis notebook - single cell studies database" href="example_analysis.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="index.html">
      
      
      
      <h1 class="site-logo" id="site-title">Single Cell Studies</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="README.html">
                    single-cell-studies
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="example_analysis.html">
   Example analysis notebook - single cell studies database
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Make plot highlighting record datasets over time
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<div class="menu-dropdown menu-dropdown-launch-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Launch interactive content">
      <i class="fas fa-rocket"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://mybinder.org/v2/gh/vals/single-cell-studies/master?urlpath=tree/201109_Make_records_figure.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Binder"
>
  

<span class="headerbtn__icon-container">
  
    <img src="_static/images/logo_binder.svg">
  </span>
<span class="headerbtn__text-container">Binder</span>
</a>

      </li>
      
      <li>
        <a href="https://colab.research.google.com/github/vals/single-cell-studies/blob/master/201109_Make_records_figure.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Launch on Colab"
>
  

<span class="headerbtn__icon-container">
  
    <img src="_static/images/logo_colab.png">
  </span>
<span class="headerbtn__text-container">Colab</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="_sources/201109_Make_records_figure.ipynb"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.ipynb</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Make plot highlighting record datasets over time</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="make-plot-highlighting-record-datasets-over-time">
<h1>Make plot highlighting record datasets over time<a class="headerlink" href="#make-plot-highlighting-record-datasets-over-time" title="Permalink to this headline">#</a></h1>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">mizani.formatters</span> <span class="kn">import</span> <span class="n">comma_format</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">plotnine</span> <span class="k">as</span> <span class="nn">p</span>

<span class="o">%</span><span class="k">pylab</span> inline
<span class="c1"># %config InlineBackend.figure_formats = [&#39;retina&#39;]</span>

<span class="kn">from</span> <span class="nn">datetime</span> <span class="kn">import</span> <span class="n">datetime</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>%pylab is deprecated, use %matplotlib inline and import the required libraries.
Populating the interactive namespace from numpy and matplotlib
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;https://nxn.se/single-cell-studies/data.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">data</span><span class="p">[</span><span class="s1">&#39;Date&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Date&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">pipe</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">to_datetime</span><span class="p">,</span> <span class="nb">format</span><span class="o">=</span><span class="s1">&#39;%Y%m</span><span class="si">%d</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">data</span><span class="p">[</span><span class="s1">&#39;Reported cells total&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Reported cells total&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="nb">float</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">KeyError</span><span class="g g-Whitespace">                                  </span>Traceback (most recent call last)
<span class="nn">File /opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pandas/core/indexes/base.py:3802,</span> in <span class="ni">Index.get_loc</span><span class="nt">(self, key, method, tolerance)</span>
<span class="g g-Whitespace">   </span><span class="mi">3801</span> <span class="k">try</span><span class="p">:</span>
<span class="ne">-&gt; </span><span class="mi">3802</span>     <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_engine</span><span class="o">.</span><span class="n">get_loc</span><span class="p">(</span><span class="n">casted_key</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">3803</span> <span class="k">except</span> <span class="ne">KeyError</span> <span class="k">as</span> <span class="n">err</span><span class="p">:</span>

<span class="nn">File /opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pandas/_libs/index.pyx:138,</span> in <span class="ni">pandas._libs.index.IndexEngine.get_loc</span><span class="nt">()</span>

<span class="nn">File /opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pandas/_libs/index.pyx:165,</span> in <span class="ni">pandas._libs.index.IndexEngine.get_loc</span><span class="nt">()</span>

<span class="nn">File pandas/_libs/hashtable_class_helper.pxi:5745,</span> in <span class="ni">pandas._libs.hashtable.PyObjectHashTable.get_item</span><span class="nt">()</span>

<span class="nn">File pandas/_libs/hashtable_class_helper.pxi:5753,</span> in <span class="ni">pandas._libs.hashtable.PyObjectHashTable.get_item</span><span class="nt">()</span>

<span class="ne">KeyError</span>: &#39;Date&#39;

<span class="n">The</span> <span class="n">above</span> <span class="n">exception</span> <span class="n">was</span> <span class="n">the</span> <span class="n">direct</span> <span class="n">cause</span> <span class="n">of</span> <span class="n">the</span> <span class="n">following</span> <span class="n">exception</span><span class="p">:</span>

<span class="ne">KeyError</span><span class="g g-Whitespace">                                  </span>Traceback (most recent call last)
<span class="n">Cell</span> <span class="n">In</span><span class="p">[</span><span class="mi">2</span><span class="p">],</span> <span class="n">line</span> <span class="mi">2</span>
<span class="g g-Whitespace">      </span><span class="mi">1</span> <span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s1">&#39;https://nxn.se/single-cell-studies/data.tsv&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="ne">----&gt; </span><span class="mi">2</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Date&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Date&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">pipe</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">to_datetime</span><span class="p">,</span> <span class="nb">format</span><span class="o">=</span><span class="s1">&#39;%Y%m</span><span class="si">%d</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">3</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Reported cells total&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">data</span><span class="p">[</span><span class="s1">&#39;Reported cells total&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">replace</span><span class="p">(</span><span class="s1">&#39;,&#39;</span><span class="p">,</span> <span class="s1">&#39;&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">map</span><span class="p">(</span><span class="nb">float</span><span class="p">)</span>

<span class="nn">File /opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pandas/core/frame.py:3807,</span> in <span class="ni">DataFrame.__getitem__</span><span class="nt">(self, key)</span>
<span class="g g-Whitespace">   </span><span class="mi">3805</span> <span class="k">if</span> <span class="bp">self</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">nlevels</span> <span class="o">&gt;</span> <span class="mi">1</span><span class="p">:</span>
<span class="g g-Whitespace">   </span><span class="mi">3806</span>     <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_getitem_multilevel</span><span class="p">(</span><span class="n">key</span><span class="p">)</span>
<span class="ne">-&gt; </span><span class="mi">3807</span> <span class="n">indexer</span> <span class="o">=</span> <span class="bp">self</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">get_loc</span><span class="p">(</span><span class="n">key</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">3808</span> <span class="k">if</span> <span class="n">is_integer</span><span class="p">(</span><span class="n">indexer</span><span class="p">):</span>
<span class="g g-Whitespace">   </span><span class="mi">3809</span>     <span class="n">indexer</span> <span class="o">=</span> <span class="p">[</span><span class="n">indexer</span><span class="p">]</span>

<span class="nn">File /opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/pandas/core/indexes/base.py:3804,</span> in <span class="ni">Index.get_loc</span><span class="nt">(self, key, method, tolerance)</span>
<span class="g g-Whitespace">   </span><span class="mi">3802</span>     <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">_engine</span><span class="o">.</span><span class="n">get_loc</span><span class="p">(</span><span class="n">casted_key</span><span class="p">)</span>
<span class="g g-Whitespace">   </span><span class="mi">3803</span> <span class="k">except</span> <span class="ne">KeyError</span> <span class="k">as</span> <span class="n">err</span><span class="p">:</span>
<span class="ne">-&gt; </span><span class="mi">3804</span>     <span class="k">raise</span> <span class="ne">KeyError</span><span class="p">(</span><span class="n">key</span><span class="p">)</span> <span class="kn">from</span> <span class="nn">err</span>
<span class="g g-Whitespace">   </span><span class="mi">3805</span> <span class="k">except</span> <span class="ne">TypeError</span><span class="p">:</span>
<span class="g g-Whitespace">   </span><span class="mi">3806</span>     <span class="c1"># If we have a listlike key, _check_indexing_error will raise</span>
<span class="g g-Whitespace">   </span><span class="mi">3807</span>     <span class="c1">#  InvalidIndexError. Otherwise we fall through and re-raise</span>
<span class="g g-Whitespace">   </span><span class="mi">3808</span>     <span class="c1">#  the TypeError.</span>
<span class="g g-Whitespace">   </span><span class="mi">3809</span>     <span class="bp">self</span><span class="o">.</span><span class="n">_check_indexing_error</span><span class="p">(</span><span class="n">key</span><span class="p">)</span>

<span class="ne">KeyError</span>: &#39;Date&#39;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;Measurement == &quot;RNA-seq&quot;&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">sort_values</span><span class="p">(</span><span class="s1">&#39;Date&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df</span><span class="p">[</span><span class="s1">&#39;Record breaker&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="kc">False</span>

<span class="n">record</span> <span class="o">=</span> <span class="mi">0</span>
<span class="k">for</span> <span class="n">n</span><span class="p">,</span> <span class="n">row</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">iterrows</span><span class="p">():</span>
    <span class="k">if</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;Reported cells total&#39;</span><span class="p">]</span> <span class="o">&gt;</span> <span class="n">record</span><span class="p">:</span>
        <span class="n">df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">n</span><span class="p">,</span> <span class="s1">&#39;Record breaker&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="kc">True</span>
        <span class="n">record</span> <span class="o">=</span> <span class="n">row</span><span class="p">[</span><span class="s1">&#39;Reported cells total&#39;</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">display</span><span class="p">(</span><span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;`Record breaker` == True&#39;</span><span class="p">));</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html">
  <div id="df-b78c7b8d-f90a-482f-86f9-8e6431b59175">
    <div class="colab-df-container">
      <div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Shorthand</th>
      <th>DOI</th>
      <th>Authors</th>
      <th>Journal</th>
      <th>Title</th>
      <th>Date</th>
      <th>bioRxiv DOI</th>
      <th>Reported cells total</th>
      <th>Organism</th>
      <th>Tissue</th>
      <th>...</th>
      <th>Cell clustering</th>
      <th>Pseudotime</th>
      <th>RNA Velocity</th>
      <th>PCA</th>
      <th>tSNE</th>
      <th>H5AD location</th>
      <th>Isolation</th>
      <th>BC --&gt; Cell ID _OR_ BC --&gt; Cluster ID</th>
      <th>Number individuals</th>
      <th>Record breaker</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>9</th>
      <td>Tang et al NMeth</td>
      <td>10.1038/NMETH.1315</td>
      <td>Fuchou Tang, Catalin Barbacioru, Yangzhou Wang...</td>
      <td>Nat Methods</td>
      <td>mRNA-Seq whole-transcriptome analysis of a sin...</td>
      <td>2009-04-06</td>
      <td>-</td>
      <td>5.0</td>
      <td>Mouse</td>
      <td>Oocyte</td>
      <td>...</td>
      <td>No</td>
      <td>No</td>
      <td>No</td>
      <td>No</td>
      <td>No</td>
      <td>NaN</td>
      <td>Pipetting (Manual picking)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>12</th>
      <td>Tang et al CSC</td>
      <td>10.1016/j.stem.2010.03.015</td>
      <td>Fuchou Tang, Catalin Barbacioru, Siqin Bao, Ca...</td>
      <td>Cell Stem Cell</td>
      <td>Tracing the Derivation of Embryonic Stem Cells...</td>
      <td>2010-05-07</td>
      <td>-</td>
      <td>34.0</td>
      <td>Mouse</td>
      <td>Embryo</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>No</td>
      <td>Yes</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>13</th>
      <td>Islam et al GR</td>
      <td>10.1101/gr.110882.110</td>
      <td>S. Islam, U. Kjallquist, A. Moliner, P. Zajac,...</td>
      <td>Genome Research</td>
      <td>Characterization of the single-cell transcript...</td>
      <td>2011-05-05</td>
      <td>-</td>
      <td>85.0</td>
      <td>Mouse</td>
      <td>Culture</td>
      <td>...</td>
      <td>Yes</td>
      <td>No</td>
      <td>NaN</td>
      <td>Yes</td>
      <td>No</td>
      <td>NaN</td>
      <td>Pipetting (Manual picking)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>17</th>
      <td>Hashimshony et al CR</td>
      <td>10.1016/j.celrep.2012.08.003</td>
      <td>Tamar Hashimshony, Florian Wagner, Noa Sher, I...</td>
      <td>Cell Reports</td>
      <td>CEL-Seq: Single-Cell RNA-Seq by Multiplexed Li...</td>
      <td>2012-08-30</td>
      <td>-</td>
      <td>96.0</td>
      <td>C elegans</td>
      <td>Embryo</td>
      <td>...</td>
      <td>No</td>
      <td>No</td>
      <td>NaN</td>
      <td>No</td>
      <td>No</td>
      <td>NaN</td>
      <td>Pipetting (Manual picking)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>24</th>
      <td>Yan et al NSMB</td>
      <td>10.1038/nsmb.2660</td>
      <td>Liying Yan, Mingyu Yang, Hongshan Guo, Lu Yang...</td>
      <td>Nat Struct Mol Biol</td>
      <td>Single-cell RNA-Seq profiling of human preimpl...</td>
      <td>2013-08-09</td>
      <td>-</td>
      <td>124.0</td>
      <td>Human</td>
      <td>Culture</td>
      <td>...</td>
      <td>Yes</td>
      <td>No</td>
      <td>NaN</td>
      <td>Yes</td>
      <td>No</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>30</th>
      <td>Islam et al NMeth</td>
      <td>10.1038/nmeth.2772</td>
      <td>Saiful Islam, Amit Zeisel, Simon Joost, Gioele...</td>
      <td>Nat Methods</td>
      <td>Quantitative single-cell RNA-seq with unique m...</td>
      <td>2013-12-22</td>
      <td>-</td>
      <td>192.0</td>
      <td>Mouse</td>
      <td>Culture</td>
      <td>...</td>
      <td>No</td>
      <td>No</td>
      <td>NaN</td>
      <td>No</td>
      <td>No</td>
      <td>NaN</td>
      <td>IFC Capture</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>31</th>
      <td>Deng et al Science</td>
      <td>10.1126/science.1245316</td>
      <td>Q. Deng, D. Ramskold, B. Reinius, R. Sandberg</td>
      <td>Science</td>
      <td>Single-Cell RNA-Seq Reveals Dynamic, Random Mo...</td>
      <td>2014-01-09</td>
      <td>-</td>
      <td>269.0</td>
      <td>Mouse</td>
      <td>Embryo</td>
      <td>...</td>
      <td>No</td>
      <td>No</td>
      <td>No</td>
      <td>Yes</td>
      <td>No</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>32</th>
      <td>Jaitin et al Science</td>
      <td>10.1126/science.1247651</td>
      <td>D. A. Jaitin, E. Kenigsberg, H. Keren-Shaul, N...</td>
      <td>Science</td>
      <td>Massively Parallel Single-Cell RNA-Seq for Mar...</td>
      <td>2014-02-14</td>
      <td>-</td>
      <td>4468.0</td>
      <td>Mouse</td>
      <td>Spleen</td>
      <td>...</td>
      <td>Yes</td>
      <td>No</td>
      <td>NaN</td>
      <td>No</td>
      <td>No</td>
      <td>NaN</td>
      <td>Sorting (FACS)</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>33</th>
      <td>Soumillon et al bioRxiv</td>
      <td>10.1101/003236</td>
      <td>Magali Soumillon, Davide Cacchiarelli, Stefan ...</td>
      <td>bioRxiv</td>
      <td>Characterization of directed differentiation b...</td>
      <td>2014-03-10</td>
      <td>10.1101/003236</td>
      <td>12832.0</td>
      <td>Human</td>
      <td>Culture</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>53</th>
      <td>Fan et al Science</td>
      <td>10.1126/science.1258367</td>
      <td>H. Christina Fan, Glenn K. Fu, Stephen P. A. F...</td>
      <td>Science</td>
      <td>Combinatorial labeling of single cells for gen...</td>
      <td>2015-02-05</td>
      <td>-</td>
      <td>15000.0</td>
      <td>Human</td>
      <td>Blood, Culture</td>
      <td>...</td>
      <td>Yes</td>
      <td>No</td>
      <td>NaN</td>
      <td>Yes</td>
      <td>No</td>
      <td>NaN</td>
      <td>Picowell deposition</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>63</th>
      <td>Macosko et al Cell</td>
      <td>10.1016/j.cell.2015.05.002</td>
      <td>Evan Z. Macosko, Anindita Basu, Rahul Satija, ...</td>
      <td>Cell</td>
      <td>Highly Parallel Genome-wide Expression Profili...</td>
      <td>2015-05-21</td>
      <td>-</td>
      <td>44808.0</td>
      <td>Human, Mouse</td>
      <td>Eye</td>
      <td>...</td>
      <td>Yes</td>
      <td>No</td>
      <td>No</td>
      <td>Yes</td>
      <td>Yes</td>
      <td>NaN</td>
      <td>Nanodroplet dilution</td>
      <td>SCP301</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>176</th>
      <td>Adamson et al Cell</td>
      <td>10.1016/j.cell.2016.11.048</td>
      <td>Britt Adamson, Thomas M. Norman, Marco Jost, M...</td>
      <td>Cell</td>
      <td>A Multiplexed Single-Cell CRISPR Screening Pla...</td>
      <td>2016-12-15</td>
      <td>-</td>
      <td>86000.0</td>
      <td>Human</td>
      <td>Culture</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Yes</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>178</th>
      <td>Dixit et al Cell</td>
      <td>10.1016/j.cell.2016.11.038</td>
      <td>Atray Dixit, Oren Parnas, Biyu Li, Jenny Chen,...</td>
      <td>Cell</td>
      <td>Perturb-Seq: Dissecting Molecular Circuits wit...</td>
      <td>2016-12-15</td>
      <td>-</td>
      <td>200000.0</td>
      <td>Human, Mouse</td>
      <td>Culture</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>No</td>
      <td>NaN</td>
      <td>Nanodroplet dilution</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>187</th>
      <td>Zheng et al NComm</td>
      <td>10.1038/ncomms14049</td>
      <td>Grace X. Y. Zheng, Jessica M. Terry, Phillip B...</td>
      <td>Nat Commun</td>
      <td>Massively parallel digital transcriptional pro...</td>
      <td>2017-01-16</td>
      <td>10.1101/065912</td>
      <td>250000.0</td>
      <td>Human, Mouse</td>
      <td>Blood</td>
      <td>...</td>
      <td>Yes</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Yes</td>
      <td>Yes</td>
      <td>NaN</td>
      <td>Nanodroplet dilution</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>311</th>
      <td>Han et al Cell</td>
      <td>10.1016/j.cell.2018.02.001</td>
      <td>Xiaoping Han, Renying Wang, Yincong Zhou, Liji...</td>
      <td>Cell</td>
      <td>Mapping the Mouse Cell Atlas by Microwell-Seq</td>
      <td>2018-02-23</td>
      <td>-</td>
      <td>400000.0</td>
      <td>Mouse</td>
      <td>Bladder, Bone marrow, Brain, Kidney, Liver, Lu...</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>399</th>
      <td>Zeisel et al Cell</td>
      <td>10.1016/j.cell.2018.06.021</td>
      <td>Amit Zeisel, Hannah Hochgerner, Peter Lönnerbe...</td>
      <td>Cell</td>
      <td>Molecular Architecture of the Mouse Nervous Sy...</td>
      <td>2018-08-09</td>
      <td>10.1101/294918</td>
      <td>509876.0</td>
      <td>Mouse</td>
      <td>Brain</td>
      <td>...</td>
      <td>Yes</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>Yes</td>
      <td>Yes</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>400</th>
      <td>Saunders et al Cell</td>
      <td>10.1016/j.cell.2018.07.028</td>
      <td>Arpiar Saunders, Evan Z. Macosko, Alec Wysoker...</td>
      <td>Cell</td>
      <td>Molecular Diversity and Specializations among ...</td>
      <td>2018-08-09</td>
      <td>10.1101/299081</td>
      <td>690000.0</td>
      <td>Mouse</td>
      <td>Brain</td>
      <td>...</td>
      <td>Yes</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>gs://h5ad/2018-08-Saunders-et-al-Cell</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>469</th>
      <td>Schmeidel et al Cell</td>
      <td>10.1016/j.cell.2018.10.022</td>
      <td>Benjamin J. Schmiedel, Divya Singh, Ariel Madr...</td>
      <td>Cell</td>
      <td>Impact of Genetic Polymorphisms on Human Immun...</td>
      <td>2018-11-15</td>
      <td>NaN</td>
      <td>1000000.0</td>
      <td>Human</td>
      <td>Blood</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>91.0</td>
      <td>True</td>
    </tr>
    <tr>
      <th>548</th>
      <td>Cao et al Nature</td>
      <td>10.1038/s41586-019-0969-x</td>
      <td>Junyue Cao, Malte Spielmann, Xiaojie Qiu, Xing...</td>
      <td>Nature</td>
      <td>The single-cell transcriptional landscape of m...</td>
      <td>2019-02-20</td>
      <td>-</td>
      <td>2058652.0</td>
      <td>Mouse</td>
      <td>Embryo</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>https://storage.googleapis.com/h5ad/2019-02-Ca...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>579</th>
      <td>Rodriques et al Science</td>
      <td>10.1126/science.aaw1219</td>
      <td>Samuel G. Rodriques, Robert R. Stickels, Aleks...</td>
      <td>Science</td>
      <td>Slide-seq: A scalable technology for measuring...</td>
      <td>2019-03-28</td>
      <td>10.1101/563395</td>
      <td>2522640.0</td>
      <td>Mouse</td>
      <td>Brain</td>
      <td>...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>https://storage.googleapis.com/h5ad/2019-03-Ro...</td>
      <td>Spatial RNA coordinates</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>True</td>
    </tr>
    <tr>
      <th>1288</th>
      <td>Cao et al Science</td>
      <td>10.1126/science.aba7721</td>
      <td>Junyue Cao, Diana R. O’Day, Hannah A. Pliner, ...</td>
      <td>Science</td>
      <td>A human cell atlas of fetal gene expression</td>
      <td>2020-11-13</td>
      <td>NaN</td>
      <td>4062980.0</td>
      <td>Human</td>
      <td>Adrenal gland, Brain, Eye, Heart, Intestine, K...</td>
      <td>...</td>
      <td>Yes</td>
      <td>Yes</td>
      <td>No</td>
      <td>Yes</td>
      <td>No</td>
      <td>https://storage.googleapis.com/h5ad/10.1126-sc...</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>121.0</td>
      <td>True</td>
    </tr>
  </tbody>
</table>
<p>21 rows × 29 columns</p>
</div>
      <button class="colab-df-convert" onclick="convertToInteractive('df-b78c7b8d-f90a-482f-86f9-8e6431b59175')"
              title="Convert this dataframe to an interactive table."
              style="display:none;">

  <svg xmlns="http://www.w3.org/2000/svg" height="24px"viewBox="0 0 24 24"
       width="24px">
    <path d="M0 0h24v24H0V0z" fill="none"/>
    <path d="M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z"/><path d="M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z"/>
  </svg>
      </button>

  <style>
    .colab-df-container {
      display:flex;
      flex-wrap:wrap;
      gap: 12px;
    }

    .colab-df-convert {
      background-color: #E8F0FE;
      border: none;
      border-radius: 50%;
      cursor: pointer;
      display: none;
      fill: #1967D2;
      height: 32px;
      padding: 0 0 0 0;
      width: 32px;
    }

    .colab-df-convert:hover {
      background-color: #E2EBFA;
      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);
      fill: #174EA6;
    }

    [theme=dark] .colab-df-convert {
      background-color: #3B4455;
      fill: #D2E3FC;
    }

    [theme=dark] .colab-df-convert:hover {
      background-color: #434B5C;
      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);
      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));
      fill: #FFFFFF;
    }
  </style>

      <script>
        const buttonEl =
          document.querySelector('#df-b78c7b8d-f90a-482f-86f9-8e6431b59175 button.colab-df-convert');
        buttonEl.style.display =
          google.colab.kernel.accessAllowed ? 'block' : 'none';

        async function convertToInteractive(key) {
          const element = document.querySelector('#df-b78c7b8d-f90a-482f-86f9-8e6431b59175');
          const dataTable =
            await google.colab.kernel.invokeFunction('convertToInteractive',
                                                     [key], {});
          if (!dataTable) return;

          const docLinkHtml = 'Like what you see? Visit the ' +
            '<a target="_blank" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'
            + ' to learn more about interactive tables.';
          element.innerHTML = '';
          dataTable['output_type'] = 'display_data';
          await google.colab.output.renderOutput(dataTable, element);
          const docLink = document.createElement('div');
          docLink.innerHTML = docLinkHtml;
          element.appendChild(docLink);
        }
      </script>
    </div>
  </div>
</div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df</span><span class="o">.</span><span class="n">columns</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Index([&#39;Shorthand&#39;, &#39;DOI&#39;, &#39;Authors&#39;, &#39;Journal&#39;, &#39;Title&#39;, &#39;Date&#39;,
       &#39;bioRxiv DOI&#39;, &#39;Reported cells total&#39;, &#39;Organism&#39;, &#39;Tissue&#39;,
       &#39;Technique&#39;, &#39;Data location&#39;, &#39;Panel size&#39;, &#39;Measurement&#39;,
       &#39;Cell source&#39;, &#39;Disease&#39;, &#39;Contrasts&#39;, &#39;Developmental stage&#39;,
       &#39;Number of reported cell types or clusters&#39;, &#39;Cell clustering&#39;,
       &#39;Pseudotime&#39;, &#39;RNA Velocity&#39;, &#39;PCA&#39;, &#39;tSNE&#39;, &#39;H5AD location&#39;,
       &#39;Isolation&#39;, &#39;BC --&gt; Cell ID _OR_ BC --&gt; Cluster ID&#39;,
       &#39;Number individuals&#39;, &#39;Record breaker&#39;],
      dtype=&#39;object&#39;)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">df</span><span class="o">.</span><span class="n">shape</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>(1649, 29)
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">p</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">figure_size</span> <span class="o">=</span> <span class="p">(</span><span class="mi">11</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>

<span class="n">fig</span> <span class="o">=</span>  <span class="p">(</span>
    <span class="n">p</span><span class="o">.</span><span class="n">ggplot</span><span class="p">(</span><span class="n">p</span><span class="o">.</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Date&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s1">&#39;Reported cells total&#39;</span><span class="p">),</span> <span class="n">df</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lightgrey&#39;</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_y_log10</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="p">[</span><span class="mf">1e1</span><span class="p">,</span> <span class="mf">1e2</span><span class="p">,</span> <span class="mf">1e3</span><span class="p">,</span> <span class="mf">1e4</span><span class="p">,</span> <span class="mf">1e5</span><span class="p">,</span> <span class="mf">1e6</span><span class="p">],</span> <span class="n">limits</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">1e7</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">comma_format</span><span class="p">())</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">theme_minimal</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">labs</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s1">&#39;Cells per study over time&#39;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Publication date&#39;</span><span class="p">)</span>
<span class="p">)</span>

<span class="n">fig</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/usr/local/lib/python3.8/dist-packages/plotnine/layer.py:401: PlotnineWarning: geom_point : Removed 358 rows containing missing values.
</pre></div>
</div>
<img alt="_images/201109_Make_records_figure_11_1.png" src="_images/201109_Make_records_figure_11_1.png" />
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;ggplot: (8792930161134)&gt;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">p</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">figure_size</span> <span class="o">=</span> <span class="p">(</span><span class="mi">11</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>

<span class="n">fig</span> <span class="o">=</span>  <span class="p">(</span>
    <span class="n">p</span><span class="o">.</span><span class="n">ggplot</span><span class="p">(</span><span class="n">p</span><span class="o">.</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Date&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s1">&#39;Reported cells total&#39;</span><span class="p">),</span> <span class="n">df</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lightgrey&#39;</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_y_log10</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="p">[</span><span class="mf">1e1</span><span class="p">,</span> <span class="mf">1e2</span><span class="p">,</span> <span class="mf">1e3</span><span class="p">,</span> <span class="mf">1e4</span><span class="p">,</span> <span class="mf">1e5</span><span class="p">,</span> <span class="mf">1e6</span><span class="p">],</span> <span class="n">limits</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">1e7</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">comma_format</span><span class="p">())</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_x_datetime</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">theme_minimal</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">labs</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s1">&#39;Cells per study over time&#39;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Publication date&#39;</span><span class="p">)</span>
<span class="p">)</span>

<span class="n">fig</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/usr/local/lib/python3.8/dist-packages/plotnine/layer.py:401: PlotnineWarning: geom_point : Removed 358 rows containing missing values.
</pre></div>
</div>
<img alt="_images/201109_Make_records_figure_12_1.png" src="_images/201109_Make_records_figure_12_1.png" />
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;ggplot: (8792961301538)&gt;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">p</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">figure_size</span> <span class="o">=</span> <span class="p">(</span><span class="mi">11</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>

<span class="n">fig</span> <span class="o">=</span>  <span class="p">(</span>
    <span class="n">p</span><span class="o">.</span><span class="n">ggplot</span><span class="p">(</span><span class="n">p</span><span class="o">.</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Date&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s1">&#39;Reported cells total&#39;</span><span class="p">),</span> <span class="n">df</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lightgrey&#39;</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;`Record breaker` == True&#39;</span><span class="p">))</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_y_log10</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="p">[</span><span class="mf">1e1</span><span class="p">,</span> <span class="mf">1e2</span><span class="p">,</span> <span class="mf">1e3</span><span class="p">,</span> <span class="mf">1e4</span><span class="p">,</span> <span class="mf">1e5</span><span class="p">,</span> <span class="mf">1e6</span><span class="p">],</span> <span class="n">limits</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">1e7</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">comma_format</span><span class="p">())</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_x_datetime</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">theme_minimal</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">labs</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s1">&#39;Cells per study over time&#39;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Publication date&#39;</span><span class="p">)</span>
<span class="p">)</span>

<span class="n">fig</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/usr/local/lib/python3.8/dist-packages/plotnine/layer.py:401: PlotnineWarning: geom_point : Removed 358 rows containing missing values.
</pre></div>
</div>
<img alt="_images/201109_Make_records_figure_13_1.png" src="_images/201109_Make_records_figure_13_1.png" />
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;ggplot: (8792928383982)&gt;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">p</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">figure_size</span> <span class="o">=</span> <span class="p">(</span><span class="mi">11</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>

<span class="n">fig</span> <span class="o">=</span>  <span class="p">(</span>
    <span class="n">p</span><span class="o">.</span><span class="n">ggplot</span><span class="p">(</span><span class="n">p</span><span class="o">.</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Date&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s1">&#39;Reported cells total&#39;</span><span class="p">),</span> <span class="n">df</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lightgrey&#39;</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;`Record breaker` == True&#39;</span><span class="p">))</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_text</span><span class="p">(</span>
        <span class="n">p</span><span class="o">.</span><span class="n">aes</span><span class="p">(</span><span class="n">label</span> <span class="o">=</span> <span class="s1">&#39;Shorthand&#39;</span><span class="p">),</span>
        <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;`Record breaker` == True&#39;</span><span class="p">),</span>
        <span class="n">size</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
        <span class="n">ha</span> <span class="o">=</span> <span class="s1">&#39;right&#39;</span>
    <span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_y_log10</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="p">[</span><span class="mf">1e1</span><span class="p">,</span> <span class="mf">1e2</span><span class="p">,</span> <span class="mf">1e3</span><span class="p">,</span> <span class="mf">1e4</span><span class="p">,</span> <span class="mf">1e5</span><span class="p">,</span> <span class="mf">1e6</span><span class="p">],</span> <span class="n">limits</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">1e7</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">comma_format</span><span class="p">())</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_x_datetime</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">theme_minimal</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">labs</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s1">&#39;Cells per study over time&#39;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Publication date&#39;</span><span class="p">)</span>
<span class="p">)</span>

<span class="n">fig</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/usr/local/lib/python3.8/dist-packages/plotnine/layer.py:401: PlotnineWarning: geom_point : Removed 358 rows containing missing values.
</pre></div>
</div>
<img alt="_images/201109_Make_records_figure_14_1.png" src="_images/201109_Make_records_figure_14_1.png" />
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;ggplot: (8792928384440)&gt;
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">p</span><span class="o">.</span><span class="n">options</span><span class="o">.</span><span class="n">figure_size</span> <span class="o">=</span> <span class="p">(</span><span class="mi">11</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>

<span class="n">fig</span> <span class="o">=</span>  <span class="p">(</span>
    <span class="n">p</span><span class="o">.</span><span class="n">ggplot</span><span class="p">(</span><span class="n">p</span><span class="o">.</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Date&#39;</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="s1">&#39;Reported cells total&#39;</span><span class="p">),</span> <span class="n">df</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lightgrey&#39;</span><span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;`Record breaker` == True&#39;</span><span class="p">))</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_text</span><span class="p">(</span>
        <span class="n">p</span><span class="o">.</span><span class="n">aes</span><span class="p">(</span><span class="n">label</span> <span class="o">=</span> <span class="s1">&#39;Shorthand&#39;</span><span class="p">),</span>
        <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;`Record breaker` == True&#39;</span><span class="p">),</span>
        <span class="n">size</span> <span class="o">=</span> <span class="mi">8</span><span class="p">,</span>
        <span class="n">adjust_text</span> <span class="o">=</span> <span class="p">{</span>
            <span class="s1">&#39;expand_points&#39;</span><span class="p">:</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">1.1</span><span class="p">),</span>
            <span class="s1">&#39;expand_text&#39;</span><span class="p">:</span> <span class="p">(</span><span class="mf">1.5</span><span class="p">,</span> <span class="mf">1.05</span><span class="p">),</span>
            <span class="s1">&#39;arrowprops&#39;</span><span class="p">:</span> <span class="p">{</span>
                <span class="s1">&#39;arrowstyle&#39;</span><span class="p">:</span> <span class="s1">&#39;-&#39;</span><span class="p">,</span>
                <span class="s1">&#39;color&#39;</span><span class="p">:</span> <span class="s1">&#39;black&#39;</span><span class="p">,</span>
                <span class="s1">&#39;lw&#39;</span><span class="p">:</span> <span class="mf">0.5</span><span class="p">,</span>
                <span class="s1">&#39;zorder&#39;</span><span class="p">:</span> <span class="mi">0</span>
            <span class="p">},</span>
            <span class="s1">&#39;autoalign&#39;</span><span class="p">:</span> <span class="s1">&#39;y&#39;</span><span class="p">,</span>
        <span class="p">}</span>
    <span class="p">)</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_y_log10</span><span class="p">(</span><span class="n">breaks</span> <span class="o">=</span> <span class="p">[</span><span class="mf">1e1</span><span class="p">,</span> <span class="mf">1e2</span><span class="p">,</span> <span class="mf">1e3</span><span class="p">,</span> <span class="mf">1e4</span><span class="p">,</span> <span class="mf">1e5</span><span class="p">,</span> <span class="mf">1e6</span><span class="p">],</span> <span class="n">limits</span> <span class="o">=</span> <span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mf">1e7</span><span class="p">),</span> <span class="n">labels</span> <span class="o">=</span> <span class="n">comma_format</span><span class="p">())</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">scale_x_datetime</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">theme_minimal</span><span class="p">()</span>
    <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">labs</span><span class="p">(</span><span class="n">title</span> <span class="o">=</span> <span class="s1">&#39;Cells per study over time&#39;</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="s1">&#39;Publication date&#39;</span><span class="p">)</span>
<span class="p">)</span>

<span class="n">fig</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output traceback highlight-ipythontb notranslate"><div class="highlight"><pre><span></span><span class="gt">---------------------------------------------------------------------------</span>
<span class="ne">PlotnineError</span><span class="g g-Whitespace">                             </span>Traceback (most recent call last)
<span class="o">&lt;</span><span class="n">ipython</span><span class="o">-</span><span class="nb">input</span><span class="o">-</span><span class="mi">13</span><span class="o">-</span><span class="mf">5e9</span><span class="n">bbd6f4e84</span><span class="o">&gt;</span> <span class="ow">in</span> <span class="o">&lt;</span><span class="n">module</span><span class="o">&gt;</span>
<span class="g g-Whitespace">      </span><span class="mi">5</span>     <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;lightgrey&#39;</span><span class="p">)</span>
<span class="g g-Whitespace">      </span><span class="mi">6</span>     <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_point</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="mi">2</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s1">&#39;black&#39;</span><span class="p">,</span> <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;`Record breaker` == True&#39;</span><span class="p">))</span>
<span class="ne">----&gt; </span><span class="mi">7</span>     <span class="o">+</span> <span class="n">p</span><span class="o">.</span><span class="n">geom_text</span><span class="p">(</span>
<span class="g g-Whitespace">      </span><span class="mi">8</span>         <span class="n">p</span><span class="o">.</span><span class="n">aes</span><span class="p">(</span><span class="n">label</span> <span class="o">=</span> <span class="s1">&#39;Shorthand&#39;</span><span class="p">),</span>
<span class="g g-Whitespace">      </span><span class="mi">9</span>         <span class="n">data</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">query</span><span class="p">(</span><span class="s1">&#39;`Record breaker` == True&#39;</span><span class="p">),</span>

<span class="nn">/usr/local/lib/python3.8/dist-packages/plotnine/geoms/geom_text.py</span> in <span class="ni">__init__</span><span class="nt">(self, mapping, data, **kwargs)</span>
<span class="g g-Whitespace">     </span><span class="mi">99</span>                 <span class="n">kwargs</span><span class="p">[</span><span class="s1">&#39;position&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">position_nudge</span><span class="p">(</span><span class="o">**</span><span class="n">nudge_kwargs</span><span class="p">)</span>
<span class="g g-Whitespace">    </span><span class="mi">100</span>         <span class="k">elif</span> <span class="ow">not</span> <span class="n">HAS_ADJUST_TEXT</span><span class="p">:</span>
<span class="ne">--&gt; </span><span class="mi">101</span>             <span class="k">raise</span> <span class="n">PlotnineError</span><span class="p">(</span>
<span class="g g-Whitespace">    </span><span class="mi">102</span>                 <span class="s2">&quot;To use adjust_text you must install the adjustText &quot;</span>
<span class="g g-Whitespace">    </span><span class="mi">103</span>                 <span class="s2">&quot;package.&quot;</span>

<span class="ne">PlotnineError</span>: &#39;To use adjust_text you must install the adjustText package.&#39;
</pre></div>
</div>
</div>
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="example_analysis.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Example analysis notebook - single cell studies database</p>
        </div>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Valentine Svensson<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>